person.rb

```diff
@@ -2,36 +2,94 @@
 require_relative './base_decorator'
 require_relative './capitalize_decorator'
 require_relative './trimmer_decorator'
+require_relative './rental'
 
 class Person < Nameable
   attr_accessor :id, :name, :age, :parent_permission
 
-  def initialize(age, name, id, parent_permission: true)
+  def initialize(age, id, parent_permission: true)
     super()
     @id = id
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
+  end
+
+  def add_rental(book, date)
+    Rental.new(date, self, book)
+  end
+
+  def correct_name
+    @name
   end
 
   def can_use_services?
     of_age? || @parent_permission
   end
 
-  def of_age?
-    @age >= 18
+  def self.create_person(persons)
+    print 'Do you want to create a student (1) or teacher (2) [Input a number]: '
+    option = gets.chomp.to_i
+
+    case option
+
+    when 1
+
+      person = create_student
+    when 2
+
+      person = create_teacher
+    else
+      puts 'Invalid input. Kindly type 1 or 2'
+    end
+    persons << person
+
+    # print "#{person.class} created successfully\n"
   end
 
-  def correct_name
-    @name
+  def self.create_student
+    print 'Classroom: '
+    classroom = gets.chomp
+
+    print 'Age: '
+    age = Interger(gets.chomp)
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Has parent permission? [Y/N]: '
+    parent_permission_user = gets.chomp
+    case parent_permission_user.downcase
+    when 'y'
+      true
+    else
+      false
+    end
+
+    person = Student.new(classroom, age, name, parent_permission)
```

Comment:
- [ ] After correcting the name error locally, i still cant create/add a STUDENT. I was getting this error. Kindly look into it 

![image](https://user-images.githubusercontent.com/63473584/198849264-83e86570-e6ef-44cc-9fa2-1f385c6f922a.png)